atlasdb-config/build.gradle

```diff
@@ -1,32 +1,41 @@
 apply from: "../gradle/publish-jars.gradle"
 apply plugin: 'org.inferred.processors'
+apply plugin: 'com.github.johnrengelman.shadow'
 
 apply from: "../gradle/shared.gradle"
 
+configurations {
+    explicitShadow
+    compile.extendsFrom(explicitShadow)
+    shadow.extendsFrom(explicitShadow)
+}
+
 dependencies {
-    compile project(':atlasdb-api')
-    compile project(':atlasdb-impl-shared')
-    compile project(':leader-election-impl')
-    compile project(':lock-impl')
+    compile group: 'com.palantir.remoting2', name: 'error-handling'
 
-    compile group: 'com.netflix.feign', name: 'feign-jackson'
-    compile (group: 'com.netflix.feign', name: 'feign-jaxrs') {
+    explicitShadow project(':atlasdb-api')
+    explicitShadow project(':atlasdb-impl-shared')
+    explicitShadow project(':leader-election-impl')
+    explicitShadow project(':lock-impl')
+
+    explicitShadow group: 'com.netflix.feign', name: 'feign-jackson'
+    explicitShadow (group: 'com.netflix.feign', name: 'feign-jaxrs') {
         exclude module: 'jsr311-api'
     }
     // versions below 8.10.0 have a bug where POST requests must have a body
-    compile group: 'com.netflix.feign', name: 'feign-okhttp'
-    compile group: 'javax.validation', name: 'validation-api'
-    compile group: 'com.palantir.config.crypto', name: 'encrypted-config-value-module'
-    compile group: 'com.palantir.tritium', name: 'tritium-lib'
+    explicitShadow group: 'com.netflix.feign', name: 'feign-okhttp'
+    explicitShadow group: 'javax.validation', name: 'validation-api'
+    explicitShadow group: 'com.palantir.config.crypto', name: 'encrypted-config-value-module'
+    explicitShadow group: 'com.palantir.tritium', name: 'tritium-lib'
 
-    compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind'
-    compile group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-yaml'
-    compile group: 'io.dropwizard', name: 'dropwizard-jackson'
-    compile group: 'com.google.code.findbugs', name: 'annotations'
+    explicitShadow group: 'com.fasterxml.jackson.core', name: 'jackson-databind'
+    explicitShadow group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-yaml'
+    explicitShadow group: 'io.dropwizard', name: 'dropwizard-jackson'
+    explicitShadow group: 'com.google.code.findbugs', name: 'annotations'
 
     // This is added so that AtlasDB clients can specify the javaAgent as a JVM argument to load jars needed for HTTP/2
     // in the boot classpath
-    runtime group: 'org.mortbay.jetty.alpn', name: 'jetty-alpn-agent', version: libVersions.jetty_alpn_agent
+    shadow group: 'org.mortbay.jetty.alpn', name: 'jetty-alpn-agent', version: libVersions.jetty_alpn_agent
```

Comment:
Slightly unrelated to this PR, but how does this interact with applications?
They removed it from http-remoting because of issues. They ask applications to do it if they want http2 support now:
https://github.com/palantir/http-remoting/pull/388